### PR TITLE
[Copy] Updates French strings to use guillemets instead of quotes

### LINKED ISSUE #12092: 🔤 French strings use quotes instead of guillemets
## 🔤 Copy

French strings use quotes instead of _guillemets_.

### Screenshot
Screen Shot 2024-12-02 at 10 00 54

<details><summary>Details</summary>
45 results - 1 file

apps/web/src/lang/fr.json:

```
    418    "07sCDh": {
    419:     "defaultMessage": "Utilisez le bouton \"Créer un processus\" pour commencer.",
    420      "description": "Instructions for adding a process item."

    546    "0jwdac": {
    547:     "defaultMessage": "Utilisez le bouton \"Créer une compétence\" pour commencer.",
    548      "description": "Instructions for adding a skill item."

    902    "2m5USi": {
    903:     "defaultMessage": "Utilisez le bouton \"Modifier\" pour commencer. ",
    904      "description": "Null message on sections for edit pool page."

   2162    "9IMcqY": {
   2163:     "defaultMessage": "En fin de compte, il revient au comité de sélection ou au gestionnaire de déterminer le nombre approprié de méthodes d’évaluation. Si vous en avez plus de {maxSteps}, sélectionnez \"évaluation supplémentaire\" en dernier ressort, puis communiquez avec notre équipe. Nous pouvons vous expliquer la façon dont ces évaluations supplémentaires peuvent être gérées sur la plateforme.",
   2164      "description": "First paragraph of second answer of the Frequently Asked Questions for logging in"

   3074    "DoZebI": {
   3075:     "defaultMessage": "\"B\" indique un niveau intermédiaire.",
   3076      "description": "Proficiency level on language requirements dialog"

   3478    "FshBb0": {
   3479:     "defaultMessage": "Commencez par ajouter un article en vous servant du bouton \"Ajouter une nouvelle compétence\" fourni à cet effet.",
   3480      "description": "Message displayed when a user has no skills in their library"

   4054    "J8Nltm": {
   4055:     "defaultMessage": "Comment vous avez pu mettre en œuvre la compétence \"{skillName}\" dans le cadre de cette expérience",
   4056      "description": "Heading for a single skill on an experience."

   5414    "Q2uL2b": {
   5415:     "defaultMessage": "Cette question aide les candidats à comprendre ce que signifie le fait dans un bassin \"pool\" de recrutement, et ce à quoi ils pourraient s’attendre en tant que candidats qui se qualifient.",
   5416      "description": "Describes the 'what to expect after admission' section of a process' advertisement."

   5978    "SfbDLA": {
   5979:     "defaultMessage": "Utilisez le bouton \"Ajouter un nouveau membre\" pour commencer.",
   5980      "description": "Instructions for adding a member to a team."

   6106    "TK0b6n": {
   6107:     "defaultMessage": "Vous pouvez vous server du bouton \"Lier une expérience\" qui sert à mettre cette compétence en évidence.",
   6108      "description": "Instructions displayed when user has experiences but hasn't featured any."

   6114    "TN9ndX": {
   6115:     "defaultMessage": "Tentez d’utiliser un autre terme ou une autre recherche pour la compétence, en vous servant du bouton \"Ajouter une nouvelle compétence\" fourni à cet effet.",
   6116      "description": "Message displayed when no skills match a users filters"

   6182    "Tl2FNA": {
   6183:     "defaultMessage": "Utilisez le bouton \"Créez une classification\" pour commencer.",
   6184      "description": "Instructions for adding a classification item."

   6729    "Wd9+xg": {
   6730:     "defaultMessage": "Utilisez le bouton \"Ajouter une nouvelle compétence\" pour commencer.",
   6731      "description": "Null message description for asset skills table."

   7213    "YiqbTc": {
   7214:     "defaultMessage": "Cette liste permet au candidat de préciser <strong>jusqu’à 3 compétences comportementales ou \"compétences relationnelles”</strong> qu’ils cherchent à améliorer activement.",
   7215      "description": "Description of a users behavioural skills to improve for admins"

   8257    "e1mxkv": {
   8258:     "defaultMessage": "Examiner et consigner les résultats de {candidateName} concernant \"{skillName}\" au \"{skillLevel}\" niveau.",
   8259      "description": "Subtitle for education requirement screening decision dialog."

   8293    "e9jncy": {
   8294:     "defaultMessage": "\"C\" est le niveau le plus élevé et il s’agit d’un niveau avancé.",
   8295      "description": "Proficiency level on language requirements dialog"

   9657    "kUkyuI": {
   9658:     "defaultMessage": "Vous pouvez utiliser le bouton \"Ajouter une compétence\" pour mettre une compétence en vedette ici.",
   9659      "description": "Secondary message to user when no skills have been attached to experience."

  10365    "o5zW6Y": {
  10366:     "defaultMessage": "Voir les définitions pour \"{skillName}\" et \"{skillLevel}\"",
  10367      "description": "Accordion title for skill and skill level header on screening decision dialog."

  10793    "qD0lyZ": {
  10794:     "defaultMessage": "Le niveau \"A\" est le niveau le moins élevé en matière de bilinguisme et indique un niveau débutant.",
  10795      "description": "Proficiency level on language requirements dialog"

  11517    "uiuMqi": {
  11518:     "defaultMessage": "Utilisez le bouton \"Ajouter une nouvelle compétence\" pour commencer.",
  11519      "description": "Null message description for essential skills table."

  12277    "z4wfGZ": {
  12278:     "defaultMessage": "Vous pouvez ajouter des points en vous servant du bouton \"Ajouter une nouvelle question \" ayant été fourni à cet effet.",
  12279      "description": "Instructions on how to add a question when there are none"
```
</details> 

## 🙋‍♀️ Proposed Implementation

Replace instances of quotes in French, for example: "string" with « string ».

## ✅ Acceptance Criteria

- [ ] French strings do not use quotes

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #12092.

## 👋 Introduction

This PR updates French strings to use _guillemets_ instead of quotes. It also adds a test to check for instances of strings in French that contain double quotes.

## 🧪 Testing

1. Verify there are no instances of `\"` in any of the fr.json files
2. Add an escaped double quote `\"`, curly opening double quote `“`, and a curly closing double quote `”`  to one of the strings in an fr.json file
3. `cd apps/web`
4. `pnpm test  src/lang/lang.test.ts`
5. Verify test fails with any of an escaped double quote `\"`, curly opening double quote `“`, or a curly closing double quote `”`

### CODE REVIEW

File: apps/web/src/lang/fr.json [hunk 1]
```diff
@@ -7211,7 +7211,7 @@
     "description": "Label for _type of resource_ fieldset in the _digital services contracting questionnaire_"
   },
   "YiqbTc": {
-    "defaultMessage": "Cette liste permet au candidat de préciser <strong>jusqu’à 3 compétences comportementales ou \"compétences relationnelles”</strong> qu’ils cherchent à améliorer activement.",
+    "defaultMessage": "Cette liste permet au candidat de préciser <strong>jusqu’à 3 compétences comportementales ou « compétences relationnelles”</strong> qu’ils cherchent à améliorer activement.",
```
Comment (vd1992) [inline apps/web/src/lang/fr.json:7214]: Oh, this string is still broken. It lacks the closing arrows

File: apps/web/src/lang/fr.json
Comment (vd1992) [inline apps/web/src/lang/fr.json:1]: A few lurking around with that curly quote from above

image

File: apps/web/src/lang/lang.test.ts
Comment (vd1992) [inline apps/web/src/lang/lang.test.ts:1]: Is it possible to test for the curly quote too? Or will testing treat it the same as regular quotes?

**Review by vd1992 — CHANGES_REQUESTED** [review]
Oh, close, those curly quotes 👀

File: apps/web/src/lang/lang.test.ts
Comment (mnigh) [reply apps/web/src/lang/lang.test.ts:1]: Great catch. d71c938e29b4f0e7cb9489e8788a9e517f5fb24a

**Review by vd1992 — APPROVED** [review]
There we go, I think it is all better now 👍

**Pull request merged**